kornia/geometry/camera/pinhole.py

```diff
@@ -61,6 +62,21 @@ def _check_valid_shape(data: torch.Tensor, data_name: str) -> bool:
             )
         return True
 
+    @staticmethod
+    def _check_consistent_device(data_iter: Iterable[torch.Tensor]) -> bool:
```

Comment:
actually this could be part of the new testing api
`KORNIA_CHECK_SAME_DEVICES`

https://github.com/kornia/kornia/blob/master/kornia/testing/__init__.py#L208